src/components/topics/TopicElementsList.vue

```diff
@@ -209,26 +216,31 @@ onMounted(() => {
                   :on-click="() => handleRemoveDataset(group, index)"
                 />
               </template>
-              <template #datasetContent="{ dataset }">
+              <template #elementContent="{ element }">
                 <!-- eslint-disable-next-line vue/no-v-html -->
-                <div v-html="fromMarkdown(dataset.purpose)"></div>
+                <div v-html="fromMarkdown(element.description)"></div>
                 <TopicDatasetCard
-                  v-if="dataset.id"
-                  :dataset-properties="dataset"
-                  :dataset-content="datasetsContent.get(dataset.id)"
+                  v-if="element.element?.id"
+                  :element="element"
+                  :dataset-content="datasetsContent.get(element.element.id)"
                 />
                 <div class="fr-grid-row">
                   <a
-                    v-if="!isAvailable(dataset.availability) && !isEdit"
+                    v-if="
+                      !isAvailable(element.extras[useSiteId()]?.availability) &&
```

Comment:
Could these sort of checks be extracted into functions in the model? Feels brittle to have such code sprinkled around.